inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/project/KnowledgeBaseIriPanel.java

```diff
@@ -0,0 +1,177 @@
+
+        // OnChange update the model with corresponding iris
+        iriSchemaChoice.setChangeHandler(new ISelectionChangeHandler<SchemaProfile>()
+        {
+            private static final long serialVersionUID = 1653808650286121732L;
+
+            @Override
+            public void onSelectionChanged(AjaxRequestTarget target, SchemaProfile bean)
+            {
+                classField.setModelObject(bean.getClassIri().stringValue());
+                subclassField.setModelObject(bean.getSubclassIri().stringValue());
+                typeField.setModelObject(bean.getTypeIri().stringValue());
+                descriptionField.setModelObject(bean.getDescriptionIri().stringValue());
+
+                target.add(comboBoxWrapper, iriSchemaChoice);
+            }
+        });
+
+        add(iriSchemaChoice);
+
+    }
+
+    private ComboBox<String> buildComboBox(String id, IModel<IRI> model, List<IRI> iris)
+    {
+        // Only set model object if it has not been initialized yet
+        if (model.getObject() == null) {
+            model.setObject(iris.get(0));
+        }
+
+        List<String> choices = iris.stream().map(IRI::stringValue).collect(Collectors.toList());
+
+        IModel<String> adapter = new LambdaModelAdapter<String>(
+            () -> model.getObject().stringValue(),
+            str -> model.setObject(SimpleValueFactory.getInstance().createIRI(str)));
+
+        ComboBox<String> comboBox = new ComboBox<String>(id, adapter, choices);
+        comboBox.add(LambdaBehavior.onConfigure(cb -> cb
+                .setEnabled(SchemaProfile.CUSTOMSCHEMA.equals(selectedSchemaProfile.getObject()))));
+        comboBox.setOutputMarkupId(true);
+        comboBox.setRequired(true);
+        comboBox.add(Validators.IRI_VALIDATOR);
+        comboBox.add(new LambdaAjaxFormComponentUpdatingBehavior("change", t -> {
+            // Do nothing just update the model values
+        }));
+        return comboBox;
+    }
+
+    private SchemaProfile checkSchemaProfile(KnowledgeBase kb)
+    {
+        SchemaProfile[] profiles = SchemaProfile.values();
+        for (int i = 0; i < profiles.length; i++) {
+            // Check if kb has a known schema profile
+            if (equalsSchemaProfile(profiles[i], kb.getClassIri(), kb.getSubclassIri(),
+                    kb.getTypeIri(), kb.getDescriptionIri())) {
+                return profiles[i];
+            }
+        }
+        // If the iris don't represent a known schema profile , return CUSTOM
+        return SchemaProfile.CUSTOMSCHEMA;
+    }
+
+    /**
+     * Compares a schema profile to given IRIs. Returns true if the IRIs are the same as in the
+     * profile
+     */
+    private boolean equalsSchemaProfile(SchemaProfile profile, IRI classIri, IRI subclassIri,
+            IRI typeIri, IRI descriptionIri)
+    {
+        if (profile.getClassIri().equals(classIri) && profile.getSubclassIri().equals(subclassIri)
```

Comment:
You can just return the result of the boolean expression, instead of using an `if`.